containers/daap-delete-data-product/tests/unit/mock_test.py

```diff
@@ -0,0 +1,161 @@
+
+    def test_success(
+        self,
+        s3_client,
+        create_glue_tables,
+        create_raw_and_curated_data,
+        data_product_name,
+        glue_client,
+        event,
+        fake_context,
+    ):
+        # Assert we have the correct number of database tables created
+        tables = glue_client.get_tables(DatabaseName=data_product_name)
+        assert len(tables["TableList"]) == 3
+
+        # Assert we have the correct number of metadata files to begin with
+        prefix = f"{data_product_name}/"
+        response = s3_client.list_objects_v2(
+            Bucket=os.getenv("METADATA_BUCKET"),
+            Prefix=prefix,
+        )
+        assert response.get("KeyCount") == 12
+
+        # Assert we have the correct number of metadata files to begin with
+        prefix = f"raw/{data_product_name}/"
+        response = s3_client.list_objects_v2(
+            Bucket=os.getenv("RAW_DATA_BUCKET"),
+            Prefix=prefix,
+        )
+        assert response.get("KeyCount") == 30
+
+        # Assert we have the correct number of metadata files to begin with
+        prefix = f"curated/{data_product_name}/"
+        response = s3_client.list_objects_v2(
+            Bucket=os.getenv("CURATED_DATA_BUCKET"),
+            Prefix=prefix,
+        )
+        assert response.get("KeyCount") == 30
+
+        ######################################################################
+        with patch("glue_utils.glue_client", glue_client):
+            # Call the handler
+            delete_data_product.handler(event, fake_context)
+
+            # Assert that all metadata files have been deleted
+            prefix = f"{data_product_name}/"
+            response = s3_client.list_objects_v2(
+                Bucket=os.getenv("METADATA_BUCKET"),
+                Prefix=prefix,
+            )
+            assert response.get("KeyCount") == 0
+
+            # Assert that raw files have been deleted
+            prefix = f"raw/{data_product_name}/"
+            response = s3_client.list_objects_v2(
+                Bucket=os.getenv("RAW_DATA_BUCKET"),
+                Prefix=prefix,
+            )
+            assert response.get("KeyCount") == 0
+
+            # Assert that curated files have been deleted
+            prefix = f"curated/{data_product_name}/"
+            response = s3_client.list_objects_v2(
+                Bucket=os.getenv("CURATED_DATA_BUCKET"),
+                Prefix=prefix,
+            )
+            assert response.get("KeyCount") == 0
+
+            with pytest.raises(glue_client.exceptions.EntityNotFoundException) as exc:
+                glue_client.get_database(Name=data_product_name)
+            assert (
+                exc.value.response["Error"]["Message"]
+                == f"Database {data_product_name} not found."
+            )
+            assert exc.value.response["Error"]["Code"] == "EntityNotFoundException"
+
+    def test_metadata_does_not_exist(self, fake_event, fake_context):
```

Comment:
I would name this something like `test_400_error_when_metadata_does_not_exist` or something along those lines - just to be clear that it's testing the case when metadata doesn't exist *before* the call, rather than testing that metadata is removed *by* the call